### PR TITLE
Refactor: Implement Cross Order Logic from Lean

### PR DESCRIPTION
**Pull request by Romazes**
#### Description

This pull request aims to refactor our existing codebase to incorporate the base logic of cross order from Lean, enhancing the efficiency and maintainability of our system.

#### Related Pull Request
- https://github.com/QuantConnect/Lean/pull/8051

#### Related Issue

N/A

#### Motivation and Context

Incorporating the base logic of cross order from Lean provides a standardized and efficient approach to order processing. By integrating this logic into our system, we aim to streamline operations, improve scalability, and enhance compatibility with industry standards.

#### Requires Documentation Change

N/A

#### How Has This Been Tested?

Specific unit tests have been added to the codebase to ensure the correct implementation and behavior of the cross order logic. These tests cover various scenarios and edge cases to validate the robustness and reliability of the refactored code.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: QuantConnect.TradierBrokerage/TradierBrokerage.cs [hunk 1]
```diff
@@ -1285,9 +1248,9 @@ private void ProcessPotentiallyUpdatedOrder(TradierCachedOpenOrder cachedOrder,
             if (updatedOrder.RemainingQuantity != cachedOrder.Order.RemainingQuantity
              || ConvertStatus(updatedOrder.Status) != ConvertStatus(cachedOrder.Order.Status))
             {
+                var leanOrderStatus = ConvertStatus(updatedOrder.Status);
                 // get original QC order by brokerage id
-                Order qcOrder;
-                if (!_zeroCrossingOrdersByTradierClosingOrderId.TryGetValue(updatedOrder.Id, out qcOrder))
+                if (!TryGetOrRemoveCrossZeroOrder(updatedOrder.Id.ToStringInvariant(), leanOrderStatus == OrderStatus.Filled, out var qcOrder))
```
Comment (Martin-Molinero) [inline QuantConnect.TradierBrokerage/TradierBrokerage.cs:1253]: Move `leanOrderStatus == OrderStatus.Filled,` internally, so it's easier on clients/consumers

File: QuantConnect.TradierBrokerage/TradierBrokerage.cs [hunk 2]
```diff
@@ -1324,83 +1287,17 @@ private void ProcessPotentiallyUpdatedOrder(TradierCachedOpenOrder cachedOrder,
...
-
-                OnOrderEvent(fill);
             }
 
             // remove from open orders since it's now closed
             if (OrderIsClosed(updatedOrder))
             {
                 _filledTradierOrderIDs.Add(updatedOrder.Id);
-                _cachedOpenOrdersByTradierOrderID.TryRemove(updatedOrder.Id, out cachedOrder);
+                _cachedOpenOrdersByTradierOrderID.TryRemove(updatedOrder.Id, out _);
```
Comment (Martin-Molinero) [inline QuantConnect.TradierBrokerage/TradierBrokerage.cs:1300]: `Remove` instead of `TryRemove`, review it I think there's an overload that doesn't return `out`

Comment (Romazes) [reply QuantConnect.TradierBrokerage/TradierBrokerage.cs:1300]: The `ConcurrentDictionary` doesn't have `Remove()` explicitly.
image

We can cast to `IDictionary<TKey, TValue>` collection to call `Remove()`. Does it make sense ?
image

**Review by Martin-Molinero — APPROVED** [review]
Looks good ty!
